akka-http-core/src/main/scala/akka/http/impl/engine/client/pool/NewHostConnectionPool.scala

```diff
@@ -427,6 +420,13 @@ private[client] object NewHostConnectionPool {
               currentTimeoutId = -1
             }
         }
+        private def hasNoEntityStream(response: HttpResponse): Boolean =
+          response.entity match {
+            case _: HttpEntity.Strict        => true
+            case HttpEntity.Default(_, _, e) => e == Source.empty
```

Comment:
The former `requestContext.request.method == HttpMethods.HEAD` has been replace by this clause. This is somewhat delicate business as we depend on equality to work, which it only does if the body stream is not decorated somewhere else.

Reply:
After all this is an optimization: for HEAD requests we want to be able to make quick progress even if the user doesn't subscribe entities (quickly enough).

In the worst case, that optimization doesn't work any more and we fall back to pre-10.2.1 times.